### PR TITLE
Add zip handling

### PR DESCRIPTION
**Pull request by Josh-XT**
This pull request adds support for converting ZIP files to markdown by extracting and converting all contained files. The most important changes include updating the documentation, importing necessary modules, implementing the `ZipConverter` class, registering the new converter, and adding tests for ZIP file processing. This uses the same logic that we use for handling zip files in [AGiXT](https://github.com/Josh-XT/AGiXT) and is very useful.

### ZIP file conversion support:

* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R15): Updated to include ZIP file support in the list of supported formats.
* [`src/markitdown/_markitdown.py`](diffhunk://#diff-141141ccfa1a8ccf158e920bbe87f2f79731875aeb5fceb6c9bf1b34ac4dcf8fR15): Imported the `zipfile` module to handle ZIP file operations.

### Implementation of `ZipConverter`:

* [`src/markitdown/_markitdown.py`](diffhunk://#diff-141141ccfa1a8ccf158e920bbe87f2f79731875aeb5fceb6c9bf1b34ac4dcf8fR841-R922): Added the `ZipConverter` class to handle the extraction and conversion of ZIP file contents.
* [`src/markitdown/_markitdown.py`](diffhunk://#diff-141141ccfa1a8ccf158e920bbe87f2f79731875aeb5fceb6c9bf1b34ac4dcf8fR966): Registered the `ZipConverter` in the `__init__` method to enable its use in the conversion process.
* [`src/markitdown/_markitdown.py`](diffhunk://#diff-141141ccfa1a8ccf158e920bbe87f2f79731875aeb5fceb6c9bf1b34ac4dcf8fR1122-R1123): Updated the `_convert` method to include the list of converters for nested processing.

### Testing:

* [`tests/test_markitdown.py`](diffhunk://#diff-d0f63ab9bbd7a09749911e1796cfb0f352250213306d60176e945a6b6bc793abR147-R152): Added tests to verify the correct processing of ZIP files and their contents.

### CODE REVIEW

**Comment by Josh-XT** [comment]
@microsoft-github-policy-service agree

File: src/markitdown/_markitdown.py [hunk 1]
```diff
@@ -837,6 +838,88 @@ def _get_mlm_description(self, local_path, extension, client, model, prompt=None
         return response.choices[0].message.content
 
 
+class ZipConverter(DocumentConverter):
+    """Converts ZIP files to markdown by extracting and converting all contained files."""
```
Comment (gagb) [inline src/markitdown/_markitdown.py:842]: Can you please expand the docstring with the example format it generates?

Comment (Josh-XT) [reply src/markitdown/_markitdown.py]: Done, is that better?

Comment (gagb) [reply src/markitdown/_markitdown.py]: Amazing 🤌🤌🤌

**Review by gagb — APPROVED** [review]

**Pull request merged**